src/Symfony/Component/Workflow/Tests/MarkingTest.php

```diff
@@ -22,24 +22,70 @@ public function testMarking()
 
         $this->assertTrue($marking->has('a'));
         $this->assertFalse($marking->has('b'));
-        $this->assertSame(['a' => 1], $marking->getPlaces());
+        $this->assertPlaces(['a' => 1], $marking);
 
         $marking->mark('b');
 
         $this->assertTrue($marking->has('a'));
         $this->assertTrue($marking->has('b'));
-        $this->assertSame(['a' => 1, 'b' => 1], $marking->getPlaces());
+        $this->assertPlaces(['a' => 1, 'b' => 1], $marking);
 
         $marking->unmark('a');
 
         $this->assertFalse($marking->has('a'));
         $this->assertTrue($marking->has('b'));
-        $this->assertSame(['b' => 1], $marking->getPlaces());
+        $this->assertPlaces(['b' => 1], $marking);
 
         $marking->unmark('b');
 
         $this->assertFalse($marking->has('a'));
         $this->assertFalse($marking->has('b'));
-        $this->assertSame([], $marking->getPlaces());
+        $this->assertPlaces([], $marking);
+
+        $marking->mark('a');
+        $this->assertPlaces(['a' => 1], $marking);
+
+        $marking->mark('a');
+        $this->assertPlaces(['a' => 2], $marking);
+
+        $marking->unmark('a');
+        $this->assertPlaces(['a' => 1], $marking);
+
+        $marking->unmark('a');
+        $this->assertPlaces([], $marking);
+    }
+
+    public function testGuardNotMarked()
+    {
+        $marking = new Marking([]);
+
+        $this->expectException(\LogicException::class);
+        $this->expectExceptionMessage('The place "a" is not marked.');
+        $marking->unmark('a');
+    }
+
+    public function testGuardNotNbTokenLowerThanZero()
+    {
+        $marking = new Marking(['a' => 1]);
+
+        $this->expectException(\LogicException::class);
+        $this->expectExceptionMessage('The place "a" could not contain a negative token number.');
+        $marking->unmark('a', 2);
+    }
+
+    public function testGuardNotNbTokenEquals0()
```

Comment:
I suggest renaming those tests to make their intent clearer:

- they don't mention that they cover the `unmark` test (and we might need a similar test cover `->mark('a', 0)`)
- `GuardNotNbTokenLowerThanZero` could be easier to understand if you describe it as unmarking more token than the current state.